starlite/_signature/models/attrs_signature_model.py

```diff
@@ -0,0 +1,343 @@
+    import attrs
+    import cattrs
+except ImportError as e:
+    raise MissingDependencyException("attrs is not installed") from e
+
+if TYPE_CHECKING:
+    from starlite._signature.parsing import ParsedSignatureParameter
+    from starlite.plugins import PluginMapping
+
+key_re = re.compile("@ attribute (.*)|'(.*)'")
+
+__all__ = ("AttrsSignatureModel",)
+
+try:
+    import pydantic
+
+    def _structure_base_model(value: Any, cls: type[pydantic.BaseModel]) -> pydantic.BaseModel:
+        if isinstance(value, pydantic.BaseModel):
+            return value
+        return cls(**value)
+
+    pydantic_hooks: list[tuple[type[Any], Callable[[Any, type[Any]], Any]]] = [
+        (pydantic.BaseModel, _structure_base_model),
+    ]
+except ImportError:
+    pydantic_hooks = []
+
+
+def _pass_through_structure_hook(value: Any, _: type[Any]) -> Any:
+    return value
+
+
+def _pass_through_unstructure_hook(value: Any) -> Any:
+    return value
+
+
+def _structure_datetime(value: Any, cls: type[datetime]) -> datetime:
+    if isinstance(value, datetime):
+        return value
+
+    try:
+        return cls.fromtimestamp(float(value))
+    except (ValueError, TypeError):
+        pass
+
+    return parse(value)
+
+
+def _structure_date(value: Any, cls: type[date]) -> date:
+    if isinstance(value, date) and not isinstance(value, datetime):
+        return value
+
+    if isinstance(value, (float, int, Decimal)):
+        return cls.fromtimestamp(float(value))
+
+    dt = _structure_datetime(value=value, cls=datetime)
+    return cls(year=dt.year, month=dt.month, day=dt.day)
+
+
+def _structure_time(value: Any, cls: type[time]) -> time:
+    if isinstance(value, time):
+        return value
+
+    if isinstance(value, str):
+        return cls.fromisoformat(value)
+
+    dt = _structure_datetime(value=value, cls=datetime)
+    return cls(hour=dt.hour, minute=dt.minute, second=dt.second, microsecond=dt.microsecond, tzinfo=dt.tzinfo)
+
+
+def _structure_timedelta(value: Any, cls: type[timedelta]) -> timedelta:
+    if isinstance(value, timedelta):
+        return value
+    if isinstance(value, (float, int, Decimal)):
+        return cls(seconds=int(value))
+    return cls(seconds=timeparse(value))
+
+
+def _structure_decimal(value: Any, cls: type[Decimal]) -> Decimal:
+    return cls(str(value))
+
+
+def _structure_path(value: Any, cls: type[PurePath]) -> PurePath:
+    return cls(str(value))
+
+
+def _structure_uuid(value: Any, cls: type[UUID]) -> UUID:
+    return value if isinstance(value, UUID) else cls(str(value))
+
+
+def _structure_multidict(value: Any, cls: type[MultiDict]) -> MultiDict:
+    return cls(value)
+
+
+def _structure_str(value: Any, cls: type[str]) -> str:
+    # see: https://github.com/python-attrs/cattrs/issues/26#issuecomment-358594015
+    if value is None:
+        raise ValueError
+    return cls(value)
+
+
+hooks: list[tuple[type[Any], Callable[[Any, type[Any]], Any]]] = [
+    (ASGIConnection, _pass_through_structure_hook),
```

Comment:
`ParsedSignatureParameter.should_skip_validation` is `True` if the parameter name is in `{"request", "socket", "scope", "receive", "send"}`. That causes the type of the model attribute to be set to `Any`:

```py
            elif parameter.should_skip_validation:
                attribute = attr.attrib(type=Any)
```

Does that mean that it is impossible for (e.g.) `Request` to actually be given as a type to an attribute on the signature model? E.g.,

```py
@get()
def my_handler(request: Request) -> None:
    ...
```

In other words, my understanding is that the attribute representing `request` on the signature model is given the type of `Any`, so that mapping of the `Request` type to the passthrough hook is redundant and that param will get structured with the `Any` hook (probably the same one, is it?).

Reply:
No, this simply means that cattrs passes the value through without modifying it